### DISCUSSION TITLE
How to use the type annotation of vuePress for defaultThemeOptions in config.ts

### DISCUSSION THREAD

**Discussion by Redmomn** [vuepress/core#1423] [Q&A] [answered]
vuepress2 version:`2.0.0-beta.67`
![image](https://github.com/vuepress/vuepress-next/assets/109732988/b29094ee-868e-4a7b-afed-cabc96e16229)
the `defineUserConfig` is 0 type parameters required

in v2.0.0-beta.35,`DefaultThemeOptions` can be used as a generic parameter for `defineUserConfig`,so how to use it in  2.0.0-beta.67

**Comment by Mister-Hope** [comment, marked as answer]
Isn't the default theme function type defined itself?

**Reply by Redmomn** [reply]
我之前借鉴别人的写法直接把主题配置写到`defineUserConfig`的键值对里面了，没有用主题的配置函数，如图所示，新版本应该是不支持这个写法了，
![image](https://github.com/vuepress/vuepress-next/assets/109732988/f290aeb5-f4ac-47d4-a55b-3b826781f713)

**Comment by meteorlxy** [comment]
Please refer to the docs. It's on the top of the default theme config page:

https://v2.vuepress.vuejs.org/reference/default-theme/config.html
